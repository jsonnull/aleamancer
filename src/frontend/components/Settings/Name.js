// @flow
import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Input from '../Input'
import editable from 'frontend/hoc/editable'

type Props = {
  name: string,
  onChange: Function
}

const Editable = editable(
  styled.div`
    position: relative;
    background: transparent;
    border: none;
    border-radius: 0;
    padding-left: 0;
    border-bottom: 1px solid ${props => props.theme.borderColor};
    color: ${props => props.theme.color};
    margin: 0 0 1rem;
    width: 100%;
    &:hover {
      cursor: pointer;
    }
  `
)

const NameInput = Input.extend`
  background: transparent;
  border: none;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  color: ${props => props.theme.color};
  margin: 0 0 1rem;
  width: 100%;
  padding: 0;
`

const Name = (props: Props) => {
  return (
    <div>
      <Label>Display Name:</Label>
      <Editable
        isHovering={true}
        value={props.name}
        input={NameInput}
        onChange={props.onChange}
      />
    </div>
  )
}

export default Name
