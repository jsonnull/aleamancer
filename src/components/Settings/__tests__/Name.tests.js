// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Name from '../Name.js'

describe('Settings Name component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Name name="name" onChange={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
