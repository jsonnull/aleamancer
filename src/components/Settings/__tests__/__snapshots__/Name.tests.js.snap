// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Name component renders correctly 1`] = `
<div>
  <label
    className="sc-bdVaJa eHwgaL"
  >
    Display Name:
  </label>
  <div
    className="sc-bxivhb eiqJAG"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    name
  </div>
</div>
`;
